### ISSUE TITLE
TarTask not packing generated js.min files (Trac #1201)

### ISSUE THREAD

**Issue by phing-issues-importer** [phingofficial/phing#575]
I just found Phing and were playing around, when i experienced a strange problem...

I'm trying to jsmin some files (using uglify-js) and add them to a tar archive. But for some reason, the TarTask won't pack the js.min files.

First i thought, the TarTask is being executed earlier/faster than my ExecTask, but that doesn't seem to be the case. For debugging purposes, i've created another ExecTask that calls PHP's sleep function. While the process was sleeping, i verified, that the js.min file is within the directory, that i am going to compress. But again, the TarTask ignored the js.min file:

``` text
<exec command="uglifyjs ${jsFile}.js -nc -o ${jsFile}.min.js " dir="${jsDir}" checkreturn="true" logoutput="true" escape="true" />
```

...

``` text
<tar destfile="${project.tempdir}/${dirname}.tar">
    <fileset dir="${dirname}">
    <patternset refid="excludedDirs" />
    </fileset>
</tar>
```

Then, i just tried the JsMinTask and ran into the same problem. However, if i generate the js.min file by myself and add it to the directory that is being compressed, it works. So there's just a problem, when generating it on the fly using Phing.

Migrated from https://www.phing.info/trac/ticket/1201

``` json
{
    "status": "new", 
    "changetime": "2015-07-30T21:57:00", 
    "description": "I just found Phing and were playing around, when i experienced a strange problem...\n\nI'm trying to jsmin some files (using uglify-js) and add them to a tar archive. But for some reason, the TarTask won't pack the js.min files.\n\nFirst i thought, the TarTask is being executed earlier/faster than my ExecTask, but that doesn't seem to be the case. For debugging purposes, i've created another ExecTask that calls PHP's sleep function. While the process was sleeping, i verified, that the js.min file is within the directory, that i am going to compress. But again, the TarTask ignored the js.min file:\n\n\n{{{\n<exec command=\"uglifyjs ${jsFile}.js -nc -o ${jsFile}.min.js \" dir=\"${jsDir}\" checkreturn=\"true\" logoutput=\"true\" escape=\"true\" />\n}}}\n\n...\n\n\n{{{\n<tar destfile=\"${project.tempdir}/${dirname}.tar\">\n    <fileset dir=\"${dirname}\">\n\t<patternset refid=\"excludedDirs\" />\n    </fileset>\n</tar>\n}}}\n\nThen, i just tried the JsMinTask and ran into the same problem. However, if i generate the js.min file by myself and add it to the directory that is being compressed, it works. So there's just a problem, when generating it on the fly using Phing.", 
    "reporter": "sascha@softcreatr.de", 
    "cc": "", 
    "resolution": "", 
    "_ts": "1438293420244865", 
    "component": "phing-core", 
    "summary": "ExecTask not packing generated js.min files", 
    "priority": "tbd", 
    "keywords": "", 
    "version": "2.11.0", 
    "time": "2015-07-30T11:18:36", 
    "milestone": "Backlog", 
    "owner": "mrook", 
    "type": "defect"
}
```

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T11:20:06: 
* sascha@softcreatr.de changed attachment from "" to "build.xml"
* sascha@softcreatr.de commented:

> Phing Build File

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T11:20:26: 
* sascha@softcreatr.de changed attachment from "" to "log.txt"
* sascha@softcreatr.de commented:

> Log output

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T11:28:02: anonymous commented:

> The title is wrong - Instead of ExecTask it should have been TarTask. Sorry.

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T11:39:35: mrook commented:

> Can you attach the output of phing -debug ?

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T21:34:26: 
* sascha@softcreatr.de changed attachment from "" to "debug.log"
* sascha@softcreatr.de commented:

> Debug Log

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T21:40:11: sascha@softcreatr.de commented:

> I think i got it:
> 
> ```TarTask``` uses ```PEAR::isError()``` - I've replaced it with ```(new PEAR)->isError``` which seem to have done the trick. I'll do some more tests, but it worked this time.

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T21:42:22: sascha@softcreatr.de commented:

> Replying to [comment:3 sascha@…]:
> > I think i got it:
> > 
> > ```TarTask``` uses ```PEAR::isError()``` - I've replaced it with ```(new PEAR)->isError``` which seem to have done the trick. I'll do some more tests, but it worked this time.
> 
> Forget that. Was my fault.
> 
> It's still not working.

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T21:54:31: 
* sascha@softcreatr.de changed attachment from "" to "proper-debug.log"
* sascha@softcreatr.de commented:

> Proper debug log (the old one is rubbish)

**Comment by phing-issues-importer** [comment]
Trac update at 20150730T21:57:00: sascha@softcreatr.de commented:

> As you can see, ```+Task: tar``` doesn't add the min.js file to the archive, but a few lines later in ```+Task: delete```, the min.js file (that has been created in ```+Task: exec``` is being deleted. So it's there.

**Comment by mrook** [comment]
This should've been fixed by 9a108da.
